notebooks/name-entity-extraction.ipynb

```diff
@@ -0,0 +1,422 @@
+    "\n",
+    "In this notebook we perform **name entity extraction (NER)**  also known as token classification: we use natural language processing models to classify the words inside a prompt. \n",
+    "\n",
+    "\n",
+    "The ease of use of the `pipeline` interface lets us quickly experiment with the pre-trained models and identify which one will work best.\n",
+    "This simple interface means that it is straightforward to access the fast inference performance of the IPU on your application.\n",
+    "\n",
+    "<img src=\"images/token_classification.png\" alt=\"Widget inference on a token classification task\" style=\"width:500px;\">\n",
+    "\n",
+    "While this notebook is focused on using the model (inference), our [token_classification](token_classification.ipynb) notebook will show you how to fine tune a model for a specific task using the [`datasets`](https://huggingface.co/docs/datasets/index) package."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b81f9be",
+   "metadata": {},
+   "source": [
+    "In order to run this demo you will need to have a Poplar SDK environment enabled with PopTorch installed\n",
+    "(see the [Getting Started](https://docs.graphcore.ai/en/latest/getting-started.html) guide for your IPU system),  and Optimum Graphcore."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First of all, let's make sure your environment has the latest version of [🤗 Optimum Graphcore](https://github.com/huggingface/optimum-graphcore) available."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install \"optimum-graphcore>=0.4, <0.5\"\n",
+    "%pip install emoji==0.6.0 gradio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The location of the cache directories can be configured through environment variables or directly in the notebook:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "executable_cache_dir = os.getenv(\"POPLAR_EXECUTABLE_CACHE_DIR\", \"./exe_cache/\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## NER with transformers pipelines on the IPU\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simplest way to use a model on the IPU is to use the `pipeline` function from the `transformers` library.\n",
+    "It provides a set of models which have been validated to work on a given task. \n",
```

Comment:
I found "It provides" a bit confusing — it looks like "It" is referring to "the pipeline function". Is that correct? Or does "It" mean the transformers library?